go/vt/vttablet/tabletserver/stateful_connection.go

```diff
@@ -51,6 +51,7 @@ type StatefulConnection struct {
 	reservedProps  *Properties
 	tainted        bool
 	enforceTimeout bool
+	timeout        time.Duration
```

Comment:
not that timeout is stored with the connection, we can store the actual time when it times out than storing the timeout value.
This will simplify the GetOutdated Logic. No need to pass the timeout value there.

Reply:
I tried making some improvements in b908164. I didn't end up storing a pre-calculated timeout time like you suggested, because:

 * Outdated is driven by `pools.Numbered.timeUsed` and `StatefulConnection.timeout` (duration)
 * These values can evolve independently: 
   * `timeUsed` changes when a connection is created or returned to pool
   * `timeout` changes based on `workload`

With each independent change, the number of times we would need to recalculate a `timeoutAt` value increases. Because of that, I felt like it would be easier to continue to store `timeUsed` and `timeout` as separate values, and calculate "is connection outdated?" in just one place. 

However, I did try to get at some of your comment. Instead of passing timeout values to the `pools.Numbered`, I moved `timeUsed` to `StatefulConnection`, so it is stored next to `timeout`. The end result is that `pools.Numbered` and the logic for getting elapsed connections is simpler.

Reply:
I agree that we need to keep `timeout`, but I still think we can keep the expiryTime rather than timeUsed, and update the expiryTime based on the `updateTime` bool in `put` call.

Reply:
OK makes sense. I tried this in 6aba785d1a.